packages/visual-editor/src/utils/visualEditorFonts.ts

```diff
@@ -43,63 +43,107 @@ export const constructFontSelectOptions = (fonts: FontRegistry) => {
  * fall outside of the font's supported values
  */
 export const constructGoogleFontLinkTags = (fonts: FontRegistry): string => {
-  const preconnectTags =
-    '<link rel="preconnect" href="https://fonts.googleapis.com">\n' +
-    '<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>\n';
+  return fontLinkDataToHTML(generateFontLinkData(fonts));
+};
 
-  const fontEntries = Object.entries(fonts);
-  const linkTags: string[] = [];
+export type FontLinkData = {
+  href: string;
+  rel: string;
+  crossOrigin?: string;
```

Comment:
Enumerate the allowed strings
https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Attributes/crossorigin